src/c++/perf_analyzer/client_backend/openai/openai_client.cc

```diff
@@ -172,7 +173,11 @@ ChatCompletionClient::AsyncInfer(
     request->timer_.CaptureTimestamp(
         triton::client::RequestTimers::Kind::REQUEST_END);
     UpdateInferStat(request->timer_);
-    if (!request->is_stream_) {
+
+    // Send final response on request completion
+    // if it has not already been sent.
+    // (e.g. in the case of seeing [DONE] in streaming case)
+    if (!request->IsFinalResponseSent()) {
```

Comment:
Shouldn't this be `if (request->IsFinalResponseSent())`, since we would want this to be true?

Reply:
the check here is to avoid sending it twice if it has already been sent - the function naming isn't the best - but wanted to to follow the "Is" pattern for booleans in other places in the header.

Final Response is sent either on [DONE] or request complete signal from curl library....

suggestions on naming welcme!

I would have 

```suggestion
    if (!request->FinalResponseSent()) {
    SendFinalResponse
```

Reply:
Thanks for explaining! This looks fine, though I do love how clean that proposed code block looks too.

Reply:
+1 for confusion here. I couldn't tell the relationship between `IsFinalResponseSent()` and its calling function

Reply:
@dyastremsky , @matthewkotila 

I originally had this - the logic as part of the sendresponse internals:

Would you prefer that?

https://github.com/triton-inference-server/client/pull/748/commits/1026361f04e6e2fddd2a40e56f1e487217964957

Reply:
Both are functionally quivalant - but I thought the explicit check before sending was clearer than ignoring it ... but doesn't seem to be all that clearer :(

Reply:
I think the one where you explicitly return reads more easily to me. It's saying we've already completed another way, we don't need to send another final response.